src/swbf_fnv_hashes.cpp

```diff
@@ -2819,16 +3868,59 @@ const std::unordered_map<std::uint32_t, std::string_view> swbf_hashes{
    "ZoomRate"_fnvp,
    "ZoomTurnDivisorMax"_fnvp,
    "ZoomTurnDivisorMin"_fnvp,
-   "ZOrder"_fnvp};
+   "ZOrder"_fnvp,
+};
+
+std::mutex swbf_hashes_mutex;
+
 }
 
 std::string lookup_fnv_hash(const std::uint32_t hash)
 {
+   std::string ret_val;
+
+   std::lock_guard lock{swbf_hashes_mutex};
    const auto result = swbf_hashes.find(hash);
 
-   if (result != std::cend(swbf_hashes)) return std::string{result->second};
+   if (result != std::cend(swbf_hashes)) {
+      // this can be interesting when building a dict
+      // synced_cout::print("looked_up:", result->second, "\n");
+      ret_val = std::string{result->second};
+   }
```

Comment:
> Please review mutex handling to ensure I understood your suggestion here.
Yup, this is safe. The mutex will be locked from this line until the scope exits, protecting access to `swbf_hashes`. But we can still improve it in two ways.


The first is quite simple since we only need to hold the lock while we access `swbf_hashes` and not for the rest of the function if we put that access in it's own scope (or even function) we limit the time we're holding the lock.
```suggestion
   {
      std::lock_guard lock{swbf_hashes_mutex};

      if (const auto result = swbf_hashes.find(hash); result != std::cend(swbf_hashes)) {
         ret_val = result->second;
      }
   }
```

The second is because we're only [reading](https://en.cppreference.com/w/cpp/container#Thread_safety) from `swbf_hashes` we don't actually need exclusive ownership, we just need to know that no other thread is going to write to it while we're reading from it. This is where the change to `std::shared_mutex` I talked about before could come into play, because with that we can then use [std::shared_lock](https://en.cppreference.com/w/cpp/thread/shared_lock) which will allow us to have multiple threads reading from it without blocking on each other. But if for whatever reason a thread has to write to it (which it admittedly shouldn't be doing in the middle of extracting a file but it is still mutable global state so we should maintain thread safety for it) the `std::lock_guard` in `add_fnv_hash` will take exclusive ownership of the mutex and stop any other thread accessing `swbf_hashes` until it's finished writing to it.
```suggestion
   {
      std::shared_lock lock{swbf_hashes_mutex};

      if (const auto result = swbf_hashes.find(hash); result != std::cend(swbf_hashes)) {
         ret_val = result->second;
      }
   }
```

Another small note is that because `swbf_hashes` is no longer storing string views the explicit construction to `std::string` can be dropped as well.